devicetypes/vision-kuowei/vision-in-wall-2relays-switch.src/vision-in-wall-2relays-switch.groovy

```diff
@@ -0,0 +1,281 @@
+		mnmn: "0ALw", 
+		vid: "812dd85f-ae1e-382e-9289-15cbc7c7fd6f"
+	) {
+		capability "Health Check"
+		capability "Refresh"
+		capability "Switch"
+		capability "Configuration"
+		// This DTH uses 2 switch endpoints. Parent DTH controls endpoint 1 so please use '1' at the end of deviceJoinName
+		// Child device (isComponent : false) representing endpoint 2 will substitute 1 with 2 for easier identification.
+		fingerprint manufacturer: "0109", prod: "2017", model: "171B", deviceJoinName: "Vision 2Relays Switch 1" //zw:Ls type:1001 mfr:0109 prod:2017 model:171B ver:16.11 zwv:4.38 lib:03 cc:98 sec:5E,72,86,85,59,70,5A,7A,60,8E,73,27,25 epc:2
+		fingerprint manufacturer: "0109", prod: "2017", model: "171C", deviceJoinName: "Vision 2Relays Switch 1" //zw:Ls type:1001 mfr:0109 prod:2017 model:171C ver:25.07 zwv:4.54 lib:03 cc:98 sec:5E,72,86,85,59,70,5A,7A,60,8E,73,27,25 epc:2
+	}
+
+	preferences {
+		parameterMap.each {
+			input (title: it.name, description: it.description, type: "paragraph", element: "paragraph")
+			switch(it.type) {
+				case "boolean":
+					input(type: "paragraph", element: "paragraph", description: "Option enabled: ${it.activeDescription}\n" + "Option disabled: ${it.inactiveDescription}")
+					input(name: it.key, type: "boolean", title: "Enable", defaultValue: it.defaultValue == it.activeOption, required: false)
+					break
+					
+				case "enum":
+					input(name: it.key, title: "Select", type: "enum", options: it.values, defaultValue: it.defaultValue, required: false)
+					break
+			}	
+		}
+	}
+}
+
+def installed() {
+	/* Push main states */
+	sendEvent(name: "healthStatus", value: "online")
+	sendEvent(name: "DeviceWatch-DeviceStatus", value: "online")
+	sendEvent(name: "DeviceWatch-Enroll", value: [protocol: "cloud", scheme:"untracked"].encodeAsJson(), displayed: false)
+	/* Child device check */
+	if(!childDevices) {
+		def d = addChildDevice(
+			"smartthings",
+			"Z-Wave Binary Switch Endpoint",
+			"${device.deviceNetworkId}-child",
+			device.hubId,
+			[
+				completedSetup: true,
+				label: "${device.displayName[0..-2]}2",
+				isComponent: false
+			]
+		)
+		d.sendEvent(name: "DeviceWatch-Enroll", value: [protocol: "cloud", scheme:"untracked"].encodeAsJson(), displayed: false)
+		d.sendEvent(name: "healthStatus", value: "online")
+	}
+	// Preferences template
+	state.currentPreferencesState = [:]
+	parameterMap.each {
+		state.currentPreferencesState."$it.key" = [:]
+		state.currentPreferencesState."$it.key".value = getPreferenceValue(it)
+		def preferenceName = it.key + "Boolean"
+		settings."$preferenceName" = true
+		state.currentPreferencesState."$it.key".status = "synced"
+	}
+	firstCommand()
+}
+
+def firstCommand(){
+	def commands = []
+	commands << encap(0, zwave.configurationV1.configurationGet(parameterNumber: 0x01))
+	commands << "delay 300"			
+	commands << encap(0, zwave.multiChannelAssociationV2.multiChannelAssociationSet(groupingIdentifier: 0x01, nodeId: [0x01, 0x01]))
```

Comment:
please use `nodeId: [0, zwaveHubNodeId, 0x1]`, as the hub's node id is not *guaranteed* to be 1 and a marker needs to be included between nodes addressed simply and nodes addressed via multi-channel endpoint

Reply:
I can only be modified to nodeId: [zwaveHubNodeId, 0x1],
  because this device too old, not support 0 in marked, 
  if I set it, that will no report when button trigger.

Reply:
so, the way multichannel association (or the older multi-instance association) works (and has worked since v1) is that the node payload is first a list of node IDs, then an (optional) marker, then a list of node IDs and endpoint IDs.

The payload you've described would be sending an association to the hub and then node 1, whatever that is.

So either you want to do `[0, zwaveHubNodeId, 0x1]` or `[zwaveHubNodeId]`

Reply:
I already use [zwaveHubNodeId] in multichannel association,
only have a question is will have Err-Code:34-501(Device connection is not secure?) at Inclusion,
but I check real signal and test all function is fine,
it's ok?

Reply:
Support already start check this issue today, will update after have new information.